### ISSUE TITLE
Search: Applied Filters should persist in UI even if no matching docs

### ISSUE THREAD

**Issue by gibrown** [Automattic/jetpack#8716]
If you search and then apply some filters, those filters should persist in the UI even if a subsequent search does not match the filter anymore. Consider:

- Search for term X
- Apply filter Y
- You have one post matching
- Then change the search term so that no posts match
- The filter Y is still applied, but is not displayed because nothing is matching it. If the user removed it then some other posts may match.

This sorta leads to the user ending up in a UI deadend. The root of the problem is that the filters we display are entirely based off of what was returned from the API. We should also include any filters that have already been applied.

**Comment by eeeeevon13** [comment]
I would agree that the filters should remain visible & consistent if a user performs a search, giving them the ability to manually remove if they so wish (expanding their search).
